core/src/test/java/org/apache/iceberg/TestMergeAppend.java

```diff
@@ -92,6 +93,77 @@ public void testEmptyTableAppend() {
         statuses(Status.ADDED, Status.ADDED));
   }
 
+  @Test
+  public void testEmptyTableAppendFilesWithDifferentSpecs() {
+    Assert.assertEquals("Table should start empty", 0, listManifestFiles().size());
```

Comment:
the community is in the process of migrating away from Junit4-style tests/assertions, so please use AssertJ-style assertions for new tests, as otherwise this adds additional overhead for the migration process. See also https://iceberg.apache.org/contribute/#assertj